File: frontend/src/types/marketData.ts
export interface MarketDataFile {
  id: number
  symbol: string
  display_name: string | null
  filename: string
  row_count: number | null
  start_date: string | null
  end_date: string | null
  timeframe: string | null
  fetched_at: string
}

export interface MarketDataFetchRequest {
  symbol: string
  timeframe: string
  start_date: string
  end_date: string
  refresh: boolean
}

File: backend/app/models/market_data.py
from datetime import date, datetime

from sqlalchemy import Date, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class MarketDataFile(Base):
    __tablename__ = "market_data_files"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    symbol: Mapped[str] = mapped_column(String, nullable=False)
    filename: Mapped[str] = mapped_column(String, nullable=False, unique=True)
    start_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    end_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    row_count: Mapped[int | None] = mapped_column(Integer, nullable=True)
    timeframe: Mapped[str | None] = mapped_column(String, nullable=True)
    uploaded_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
